ESP32/main/src/gamepads/ps_3_4_controller.cpp

```diff
@@ -0,0 +1,540 @@
+{
+    #ifdef IS_PS4
+      return PS4.event.button_down.share;
+    #elif IS_PS3
+      return Ps3.event.button_down.share;
+    #else
+      return 0;
+    #endif
+}
+
+int ps_3_4_controller::is_square()
+{
+    #ifdef IS_PS4
+      return PS4.event.button_down.square;
+    #endif
+    #ifdef IS_PS3
+      return Ps3.event.button_down.square;
+    #endif
+}
+
+int ps_3_4_controller::is_circle()
+{
+    #ifdef IS_PS4
+      return PS4.event.button_down.circle;
+    #endif
+    #ifdef IS_PS3
+      return Ps3.event.button_down.circle;
+    #endif
+}
+
+int ps_3_4_controller::is_triangle()
+{
+    #ifdef IS_PS4
+      return PS4.event.button_down.triangle;
+    #endif
+    #ifdef IS_PS3
+      return Ps3.event.button_down.triangle;
+    #endif
+}
+
+int ps_3_4_controller::is_down_down()
+{
+    #ifdef IS_PS4
+      return PS4.event.button_down.down;
+    #endif
+    #ifdef IS_PS3
+      return Ps3.event.button_down.down;
+    #endif
+}
+
+int ps_3_4_controller::is_down_up()
+{
+    #ifdef IS_PS4
+      return PS4.event.button_down.up;
+    #endif
+    #ifdef IS_PS3
+      return Ps3.event.button_down.up;
+    #endif
+}
+
+int ps_3_4_controller::is_down_left()
+{
+    #ifdef IS_PS4
+      return PS4.event.button_down.left;
+    #endif
+    #ifdef IS_PS3
+      return Ps3.event.button_down.left;
+    #endif
+}
+
+int ps_3_4_controller::is_down_right()
+{
+    #ifdef IS_PS4
+      return PS4.event.button_down.right;
+    #endif
+    #ifdef IS_PS3
+      return Ps3.event.button_down.right;
+    #endif
+}
+
+int ps_3_4_controller::is_charging()
+{
+    #ifdef IS_PS4
+      return PS4.Charging();
+    #endif
+    #ifdef IS_PS3
+      return Ps3.Charging();
+    #endif
+}
+
+int ps_3_4_controller::is_connected()
+{
+    #ifdef IS_PS4
+      return PS4.isConnected();
+    #endif
+    #ifdef IS_PS3
+      return Ps3.isConnected();
+    #endif
+}
+
+int ps_3_4_controller::analog_ly()
+{
+    #ifdef IS_PS4
+      return PS4.data.analog.stick.ly;
+    #endif
+    #ifdef IS_PS3
+      return Ps3.data.analog.stick.ly;
+    #endif
+}
+
+int ps_3_4_controller::analog_lx()
+{
+    #ifdef IS_PS4
+      return PS4.data.analog.stick.lx;
+    #endif
+    #ifdef IS_PS3
+      return Ps3.data.analog.stick.lx;
+    #endif
+}
+
+int ps_3_4_controller::analog_ry()
+{
+    #ifdef IS_PS4
+      return PS4.data.analog.stick.ry;
+    #endif
+    #ifdef IS_PS3
+      return Ps3.data.analog.stick.ry;
+    #endif
+}
+
+int ps_3_4_controller::analog_rx()
+{
+    #ifdef IS_PS4
+      return PS4.data.analog.stick.rx;
+    #endif
+    #ifdef IS_PS3
+      return Ps3.data.analog.stick.rx;
+    #endif
+}
+
+int ps_3_4_controller::left()
+{
+    #ifdef IS_PS4
+      return PS4.data.button.left;
+    #endif
+    #ifdef IS_PS3
+      return Ps3.data.button.left;
+    #endif
+}
+
+int ps_3_4_controller::right()
+{
+    #ifdef IS_PS4
+      return PS4.data.button.right;
+    #endif
+    #ifdef IS_PS3
+      return Ps3.data.button.right;
+    #endif
+}
+
+/*int ps_3_4_controller::start_b()
```

Comment:
any reason why this is outcommented?

Reply:
one of the psx libs dont contain start_b function. thats why its disabled it.